### PR TITLE
feat: Added Product Page

### PR DESCRIPTION
**Pull request by parthgabani777**
- Added Product Listing Page
- Created product context

### CODE REVIEW

File: src/components/filter.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+                            onChange={(e) => {
+                                dispatchFilters({
+                                    type: "Category_History",
+                                });
+                            }}
+                        />
+                        <label htmlFor="history">History</label>
+                    </div>
+
+                    <div className="input-group">
```
Comment (prathmeshjagtap) [inline src/components/filter.js]: Instead of repeating code you can make an array of categories  and map over it

File: src/components/filter.js [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+                        <input
+                            type="radio"
+                            name="rating"
+                            id="rating1"
+                            checked={rating == 1}
+                            onChange={() => {
+                                dispatchFilters({
+                                    type: "Rating_Filter",
+                                    payload: { rating: 1 },
```
Comment (prathmeshjagtap) [inline src/components/filter.js]: you can also make an array of rating and map over it

File: src/pages/product/Product.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    const [filters, dispatchFilters] = useReducer(filterReducer, defaultValue);
+    const { categories, rating, sort, price } = filters;
+
+    const priceFilteredItems = filterPriceRange(products, price);
+    const categoryFilteredItems = filterByCategory(
+        priceFilteredItems,
+        categories
+    );
+    const ratingFilteredItems = filterByRating(categoryFilteredItems, rating);
+    const sortedItems = sortItems(ratingFilteredItems, sort);
```
Comment (prathmeshjagtap) [inline src/pages/product/Product.js]: you can try to make a compose function.

**Review by prathmeshjagtap — APPROVED** [review]
These are some of the suggestions if you have less time complete the project first and then improve  this code

File: src/utils/filter-utils.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    }
+};
+
+const filterPriceRange = (items, filterPrice) => {
+    return items.filter((item) => {
+        return parseInt(item.price) <= filterPrice;
+    });
+};
+
+const filterByRating = (items, rating) => {
```
Comment (deepak29-git) [inline src/utils/filter-utils.js:26]: you have to make another separate file for array manipulation operation  so it will look clean

**Review by deepak29-git — APPROVED** [review]

**Pull request merged**